packages/vaadin-grid/src/vaadin-grid.js

```diff
@@ -343,11 +344,19 @@ class GridElement extends ElementMixin(
   }
 
   static get observers() {
-    return ['_columnTreeChanged(_columnTree, _columnTree.*)'];
+    return [
+      '_columnTreeChanged(_columnTree, _columnTree.*)',
+      '_effectiveSizeChanged(_effectiveSize, __virtualizer, _hasData, _columnTree)'
```

Comment:
I'm wondering if we should add `isAttached` here as well? How about this scenario:

1. Remove `vaadin-grid` from the DOM
2. Set different `size`
3. Add `vaadin-grid` back to the DOM

Just guessing, I haven't actually tried this. So this might be not an issue.